src/plugins/data/public/search/expressions/esaggs.ts

```diff
@@ -67,6 +67,15 @@ interface Arguments {
   aggConfigs: string;
 }
 
+type ExpressionFunctionEsaggs = ExpressionFunctionDefinition<typeof name, Input, Arguments, Output>;
+
+// Add esaggs function to ExpressionFunctionDefinitions
+declare module '../../../../expressions/public' {
+  interface ExpressionFunctionDefinitions {
+    esaggs: ExpressionFunctionEsaggs;
+  }
+}
```

Comment:
The catch is that each plugin registering an expression function will need to do this. If the function is available on the server they will need to declare it there too. (Or if we end up moving toward `common` being a first-class citizen in plugins, they'll only have to do this once)